components/browser/menu/src/main/java/mozilla/components/browser/menu/view/ExpandableLayout.kt

```diff
@@ -0,0 +1,379 @@
+     */
+    @VisibleForTesting
+    internal var lastVisibleItemIndexWhenCollapsed: Int = Int.MAX_VALUE
+
+    /**
+     * Height of wrapped view when collapsed.
+     * Calculated once based on the position of the "isCollapsingMenuLimit" BrowserMenuItem.
+     * Capped by [parentHeight]
+     */
+    @VisibleForTesting
+    internal var collapsedHeight: Int = NOT_CALCULATED_DEFAULT_HEIGHT
+
+    /**
+     * Height of wrapped view when expanded.
+     * Calculated once based on measuredHeighWithMargins().
+     * Capped by [parentHeight]
+     */
+    @VisibleForTesting
+    internal var expandedHeight: Int = NOT_CALCULATED_DEFAULT_HEIGHT
+
+    /**
+     * Available space given by the parent.
+     */
+    @VisibleForTesting
+    internal var parentHeight: Int = NOT_CALCULATED_DEFAULT_HEIGHT
+
+    /**
+     * Whether to intercept touches while the view is collapsed.
+     * If true:
+     * - a swipe up will be intercepted and used to expand the wrapped view.
+     * - a swipe in the empty space left by the collapsed view will be intercepted
+     * and [blankTouchListener] will be called.
+     * - other touches / gestures will be left to pass through to the children.
+     */
+    @VisibleForTesting
+    internal var isCollapsed = true
+
+    /**
+     * Whether to intercept touches while the view is expanding.
+     * If true:
+     * - all touches / gestures will be intercepted.
+     */
+    @VisibleForTesting
+    internal var isExpandInProgress = false
+
+    /**
+     * Distance in pixels a touch can wander before we think the user is scrolling.
+     * (If this would be bigger than that of a child the child will react to the scroll first)
+     */
+    @VisibleForTesting
+    internal var touchSlop = ViewConfiguration.get(context).scaledTouchSlop.toFloat()
+
+    /**
+     * Y axis coordinate of the [MotionEvent.ACTION_DOWN] event.
+     * Used to calculate the distance scrolled, to know when the view should be expanded.
+     */
+    @VisibleForTesting
+    internal var initialYCoord = NOT_CALCULATED_Y_TOUCH_COORD
+
+    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
+        callParentOnMeasure(widthMeasureSpec, heightMeasureSpec)
+
+        // Avoid new separate measure calls specifically for our usecase. Piggyback on the already requested ones.
+        // Calculate our needed dimensions and collapse the menu when based on them.
+        if (isCollapsed && getOrCalculateCollapsedHeight() > 0 && getOrCalculateExpandedHeight(heightMeasureSpec) > 0) {
+            collapse()
+        }
+    }
+
+    // While this view is collapsed (not fully expanded) we want to intercept all vertical scrolls
+    // that will be used as an indicator to expand the view,
+    // while letting all simple touch events get handled by children's click listeners.
+    //
+    // Also if this view is collapsed (full height but translated) we want to treat any touch in the
+    // invisible space as a dismiss event.
```

Comment:
```suggestion
    /** While this view is collapsed (not fully expanded) we want to intercept all vertical scrolls
    * that will be used as an indicator to expand the view,
    * while letting all simple touch events get handled by children's click listeners.
    * 
    * Also if this view is collapsed (full height but translated) we want to treat any touch in the
    * invisible space as a dismiss event.
    */
```

Reply:
I've accustomed to use slash comments for purely technical details, maybe how something is implemented, details that are of no use to clients of those APIs and KDocs to explain to consumers how/when to use a particular API.
I feel like this is the case here but if you feel strongly I can change this comment.